custom_components/unifiprotect/switch.py

```diff
@@ -132,36 +148,31 @@ def device_state_attributes(self):
 
         return attrs
 
-    def turn_on(self, **kwargs):
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+    async def async_turn_on(self, **kwargs):
         """Turn the device on."""
         if self._switch_type == "record_motion":
             _LOGGER.debug("Turning on Motion Detection")
-            self.data.set_camera_recording(self._camera_id, TYPE_RECORD_MOTION)
+            await self.upv.set_camera_recording(self._camera_id, TYPE_RECORD_MOTION)
         elif self._switch_type == "record_always":
             _LOGGER.debug("Turning on Constant Recording")
-            self.data.set_camera_recording(self._camera_id, TYPE_RECORD_ALLWAYS)
+            await self.upv.set_camera_recording(self._camera_id, TYPE_RECORD_ALLWAYS)
         else:
             _LOGGER.debug("Turning on IR")
-            self.data.set_camera_ir(self._camera_id, self._ir_on_cmd)
+            await self.upv.set_camera_ir(self._camera_id, self._ir_on_cmd)
+        await self.coordinator.async_request_refresh()
 
-    def turn_off(self, **kwargs):
+    async def async_turn_off(self, **kwargs):
         """Turn the device off."""
         if self._switch_type == "ir_mode":
             _LOGGER.debug("Turning off IR")
-            self.data.set_camera_ir(self._camera_id, self._ir_off_cmd)
+            await self.upv.set_camera_ir(self._camera_id, self._ir_off_cmd)
         else:
             _LOGGER.debug("Turning off Recording")
-            self.data.set_camera_recording(self._camera_id, TYPE_RECORD_NEVER)
-
-    def update(self):
```

Comment:
The data coordinator handles updating the data so this shouldn't be implemented.  I moved the logic in here to the is_on property.